mne/epochs.py

```diff
@@ -2600,6 +2621,7 @@ def as_type(self, ch_type="grad", mode="fast"):
         """
         from .forward import _as_meg_type_inst
 
+        _check_empty(self, "as_type")
         return _as_meg_type_inst(self, ch_type=ch_type, mode=mode)
```

Comment:
Same, this could call `get_data(..., on_empty='raise')`.

It makes me think that we might want to add `get_data(..., on_missing='raise')` by default as a safety/bugfix, then we get all of these checks for free in our code base and users probably get safer behavior (arguably cannot `get_data` if there is none to get...) with the ability to do `on_empty='ignore'` if they *really* want an array of size 0